### DISCUSSION TITLE
incorporating LEAP

### DISCUSSION THREAD

**Discussion by kylechampley** [odlgroup/odl#1641] [Ideas]
Hello.  ODL is a very cool concept; I love the high-level abstraction that you are able to achieve with this package.  I am writing to see if you are interested in incorporating a tomographic reconstruction package that I've been working on with a colleague of mine called [LEAP](https://github.com/LLNL/LEAP).

LEAP employs more accurate projectors and has more features than other tomography software packages.  Let me know if you are interested and if I can be of any assistance in getting start with LEAP.

**Comment by ozanoktem** [comment]
[LEAP]( https://github.com/LLNL/LEAP) (and [XrayPhysics](https://github.com/kylechampley/XrayPhysics)) seem to be well structured libraries for X-ray imaging. We would be glad to assist any attempts at integrating LEAP with ODL, but bear in mind that we have limited resources, so the main work needs to be done by those directly involved in LEAP. Since LEAP already has python APIs, this should not be that difficult. The key component would be to map ODL geometry objects to those accepted by LEAP. Either way, I expect this to be similar to the binding code that was written ti integrate [ASTRA](https://github.com/astra-toolbox/astra-toolbox) with ODL.

**Comment by kylechampley** [comment]
Thanks for writing back.  Sure, I'd love to give it a try!  Would you please send me a link to the ASTRA integration code?  Thanks.

**Comment by ozanoktem** [comment]
Integration code for computing the 2D/3D ray transform using ASTRA (and skimage in 2D only) can be found in [https://github.com/odlgroup/odl/tree/master/odl/tomo/backends](https://github.com/odlgroup/odl/tree/master/odl/tomo/backends).

**Comment by kylechampley** [comment]
Thanks, I'll take a look.

**Reply by Emvlt** [reply]
Hello Kyle!

We are still interested in integrating LEAP as a computational backend for ODL. Is it something you are still willing to do/would be keen to do jointly with us? :)
